pkg/hubble/relay/server/server_test.go

```diff
@@ -0,0 +1,310 @@
+	"github.com/cilium/cilium/pkg/hubble/server"
+	"github.com/cilium/cilium/pkg/hubble/server/serveroption"
+	"github.com/cilium/cilium/pkg/hubble/testutils"
+	"github.com/cilium/cilium/pkg/monitor"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
+)
+
+var log *logrus.Logger
+
+func init() {
+	log = logrus.New()
+	log.SetOutput(io.Discard)
+}
+
+func noopParser(t testing.TB) *parser.Parser {
+	pp, err := parser.New(
+		log,
+		&testutils.FakeEndpointGetter{
+			OnGetEndpointInfo: func(ip netip.Addr) (endpoint v1.EndpointInfo, ok bool) {
+				endpoint, ok = endpoints[ip.String()]
+				return
+			},
+			OnGetEndpointInfoByID: func(id uint16) (endpoint v1.EndpointInfo, ok bool) {
+				return nil, false
+			},
+		},
+		&testutils.NoopIdentityGetter,
+		&testutils.NoopDNSGetter,
+		&testutils.NoopIPGetter,
+		&testutils.NoopServiceGetter,
+		&testutils.NoopLinkGetter,
+		&testutils.NoopPodMetadataGetter,
+	)
+	require.NoError(t, err)
+	return pp
+}
+
+var endpoints map[string]*testutils.FakeEndpointInfo
+
+func init() {
+	endpoints = make(map[string]*testutils.FakeEndpointInfo, 254)
+	for i := 0; i < 254; i++ {
+		ip := fake.IP(fake.WithIPv4())
+		endpoints[ip] = &testutils.FakeEndpointInfo{
+			ID:           uint64(i),
+			IPv4:         net.ParseIP(ip),
+			PodNamespace: fake.K8sNamespace(),
+			PodName:      fake.K8sPodName(),
+			Labels:       fake.K8sLabels(),
+		}
+	}
+}
+
+func getRandomEndpoint() *testutils.FakeEndpointInfo {
+	for _, v := range endpoints {
+		return v
+	}
+	return nil
+}
+
+func newHubbleObserver(t testing.TB, nodeName string, numFlows int) *observer.LocalObserverServer {
+	queueSize := numFlows
+
+	pp := noopParser(t)
+	s, err := observer.NewLocalServer(pp, log,
+		observeroption.WithMaxFlows(container.Capacity65535),
+		observeroption.WithMonitorBuffer(queueSize),
+	)
+	require.NoError(t, err)
+
+	m := s.GetEventsChannel()
+
+	for i := 0; i < numFlows; i++ {
+		tn := monitor.TraceNotifyV0{Type: byte(monitorAPI.MessageTypeTrace)}
+		src := getRandomEndpoint()
+		dst := getRandomEndpoint()
+		srcMAC, _ := net.ParseMAC(fake.MAC())
+		dstMAC, _ := net.ParseMAC(fake.MAC())
+		data := testutils.MustCreateL3L4Payload(tn,
+			&layers.Ethernet{
+				SrcMAC:       srcMAC,
+				DstMAC:       dstMAC,
+				EthernetType: layers.EthernetTypeIPv4,
+			},
+			&layers.IPv4{
+				SrcIP:    src.IPv4,
+				DstIP:    dst.IPv4,
+				Protocol: layers.IPProtocolTCP,
+			},
+			&layers.TCP{
+				SrcPort: layers.TCPPort(fake.Port()),
+				DstPort: layers.TCPPort(fake.Port()),
+				ACK:     true,
+				PSH:     i%4 == 0,
+			})
+		event := &observerTypes.MonitorEvent{
+			Timestamp: time.Unix(int64(i+1), 0),
+			NodeName:  nodeName,
+			Payload: &observerTypes.PerfEvent{
+				Data: data,
+				CPU:  0,
+			},
+		}
+		m <- event
+	}
+
+	return s
+}
+
+func newHubblePeer(t testing.TB, ctx context.Context, address string, hubbleObserver *observer.LocalObserverServer) {
+	options := []serveroption.Option{
+		serveroption.WithInsecure(),
```

Comment:
```suggestion
		serveroption.WithTransportCredentials(insecure.NewCredentials()),
```

Reply:
There is no such option: https://github.com/cilium/cilium/blob/main/pkg/hubble/server/serveroption/option.go

Reply:
Oh, sorry about that, I confused it with the grpc option :man_facepalming: